src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -433,6 +518,43 @@ const char *smbios_mainboard_sku(void)
 	return sku;
 }
 
+int fill_mainboard_smbios_type16(unsigned long *current, int *handle)
+{
+	u8 spd_index = get_spd_offset();
+	u8 spd_buffer[CONFIG_DIMM_SPD_SIZE];
+	if (read_ddr3_spd_from_cbfs(spd_buffer, spd_index) < 0) {
+		return 0;
+	}
+
+	struct smbios_type16 *t = (struct smbios_type16 *)*current;
+	int len = sizeof(struct smbios_type16) - 2;
+	memset(t, 0, sizeof(struct smbios_type16));
+
+	t->handle = *handle;
+	t->length = len;
+	t->type = SMBIOS_PHYS_MEMORY_ARRAY;
+	t->use = MEMORY_ARRAY_USE_SYSTEM;
+	t->location = MEMORY_ARRAY_LOCATION_SYSTEM_BOARD;
+	t->maximum_capacity = 4 * 1024 * 1024; // 4GB (in kB) due to board design
```

Comment:
@finus16 we have platforms with 2GB variants. Check how the memory configuration is determined.